src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -932,14 +910,52 @@ protected override void EndProcessing()
             {
                 // Load the source if they want to load from a file
                 if (String.Equals(ParameterSetName, "FromPath", StringComparison.OrdinalIgnoreCase) ||
-                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase))
+                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase)
+                   )
                 {
-                    this.sourceCode = "";
-                    foreach (string file in paths)
+                    if (paths.Length == 1)
+                    {
+                        sourceCode = File.ReadAllText(paths[0]);
+                    }
+                    else
                     {
-                        this.sourceCode += System.IO.File.ReadAllText(file) + "\n";
+
+                        long initLength = 0;
+
+                        foreach (string file in paths)
+                        {
+                            FileInfo f = new FileInfo(file);
+                            initLength = f.Length;
+                        }
+
+                        if (initLength < int.MaxValue)
+                        {
+                            StringBuilder sb = new StringBuilder((int)initLength);
+
+                            foreach (string file in paths)
+                            {
+                                foreach (string line in File.ReadAllLines(file))
+                                {
+                                    sb.AppendLine(line);
+                                }
+                            }
+
+                            sourceCode = sb.ToString();
+                        }
+                        else
+                        {
+                            sourceCode = "";
```

Comment:
It was my understanding that ReadAllText() uses StringBuilder internally.  Do we know that it correctly handles strings larger than MaxInt characters?

Reply:
My bad. I removed the code.